### DISCUSSION TITLE
Documentation for theme CSS overrides

### DISCUSSION THREAD

**Discussion by johnthagen** [squidfunk/mkdocs-material#6081] [Q&A] [answered]
At

- https://github.com/squidfunk/mkdocs-material/issues/6061#issuecomment-1730194281

it was shown how to override specific CSS settings in the base `slate` theme. Is there a place that this is documented so that if someone wanted to tweak other settings, they would know where to look (like some kind of reference)?

The docs mention `extra.css` (which we have used) but not what all of the theme options are:

- https://squidfunk.github.io/mkdocs-material/customization/#additional-css

If this is supported, could something be added to the docs about this? For example, if someone wanted to tweak the title foreground color, background color, etc? Perhaps a link to the source files in the repo that should be read to learn about the CSS options?

**Comment by squidfunk** [comment, marked as answer]
Thanks for asking. We mention it in our [custom color guide](https://squidfunk.github.io/mkdocs-material/setup/changing-the-colors/#custom-colors) that links to the [color definitions](https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/assets/stylesheets/main/_colors.scss), i.e., the variables which you can override. There's also a [related example](https://mkdocs-material.github.io/examples/custom-colors/) that you can inspect and download, that shows you how to apply custom colors.

That being said, it is impossible for us to provide a guide how to change every aspect of Material for MkDocs, e.g., header, footer, you name it, because there are just so many things that you can change (= everything). We will provide more examples in the future, but in the end, if you want to override specific parts of the CSS, you should inspect the element with your browsers console to learn which class you need to apply. We might create a tutorial for this in the future, but please understand that you'd we presume a basic knowledge of CSS for extensive customization.

**Reply by johnthagen** [reply]
I was playing around with 

```cs
[data-md-color-scheme=slate] {
    --md-default-fg-color: hsla(var(--md-hue), 15%, 90%, 1);
}
```

in my `extra.css` and noticed that it doesn't change `h1` color, but it does for everything else in the document (like `h2`s).

When I inspect the header:

```css
.md-typeset h1 {
    color: var(--md-default-fg-color--light);
    font-size: 2em;
    line-height: 1.3;
    margin: 0 0 1.25em;
}
```

When I disable `color` on that, then the title shows up in the correct brighter text like everything else.

Is it possible that the `h1` color styling is hard coded to light mode, even for `slate` theme? Sorry if as I am in investigating this further this is turning into a possible bug report.

I can create a more detailed issue if I'm not simply misunderstanding how the CSS is supposed to work here.

**Reply by johnthagen** [reply]
Here is a repo where you can see the issue, if this is in fact an issue:

- https://github.com/johnthagen/python-blueprint/tree/material-mkdocs-9.4

<img width="605" alt="Screenshot 2023-09-25 at 8 23 32 AM" src="https://github.com/squidfunk/mkdocs-material/assets/10340167/abd7e366-0b62-45f5-9255-31bae389d03c">

**Reply by squidfunk** [reply]
> Is it possible that the h1 color styling is hard coded to light mode, even for slate theme? Sorry if as I am in investigating this further this is turning into a possible bug report.

No. It's set to whatever value `--md-default-fg-color--light` has, which is defined for both color schemes.

**Reply by sjrothfuss** [reply]
I see the same, changing `h1` color in stylesheets/extra.css does nothing, changing `h2` color works.

**Reply by squidfunk** [reply]
Are you using `.md-typeset h1` as a selector? You must use a selector of same or greater [specificity](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity).

**Reply by sjrothfuss** [reply]
Sorry to not engage on that point. Yes, adding `.md-typeset` changes `h1` and that is not required for `h2`, i.e. this turns both headers red.

docs/stylesheets/extra.css:
```css
.md-typeset h1 {
    color: red;
}

h2 {
    color: red;
}
```

I think I've just muddied the water. Apologies.
